pkg/controller/agent/pod.go

```diff
@@ -121,5 +121,6 @@ func createDataVolume(params Params) volume.VolumeLike {
 }
 
 func certificatesDir(association commonv1.Association) string {
-	return fmt.Sprintf("/mnt/elastic-internal/%s-certs", association.AssociatedType())
+	ref := association.AssociationRef()
+	return fmt.Sprintf("/mnt/elastic-internal/%s/%s/%s/certs", association.AssociationType(), ref.Namespace, ref.Name)
```

Comment:
Maybe something like `/mnt/elastic-internal/elasticsearch-association/namespace/name` would make things a bit more explicit when looking at the file system?